examples/list.rs

```diff
@@ -202,80 +188,159 @@ fn run_app<B: Backend>(
 }
 
 fn ui(f: &mut Frame, app: &mut App) {
+    // Create a space for header
+    let vertical = Layout::vertical([Constraint::Length(2), Constraint::Percentage(100)]);
+    let [header_area, rest_area] = f.size().split(&vertical);
+
     // Create two chunks with equal horizontal screen space
-    let horizontal = Layout::horizontal([Constraint::Percentage(50), Constraint::Percentage(50)]);
-    let [item_list_area, event_list_area] = f.size().split(&horizontal);
+    let horizontal = Layout::horizontal([Constraint::Percentage(50), Constraint::Max(15)]);
+    let [item_list_area, event_list_area] = rest_area.split(&horizontal);
+
+    // Create two chunks with equal vertical screen space. One for the list and the other for the
+    // info block.
+    let vertical = Layout::vertical([Constraint::Percentage(50), Constraint::Percentage(50)]);
+    let [upper_item_list_area, lower_item_list_area] = item_list_area.split(&vertical);
+
+    let header = Block::default()
+        .title("Use j k or ↓ ↑ to move, h or ← to unselect, l → to change status.")
+        .title_alignment(Alignment::Center);
+    f.render_widget(header, header_area);
+
+    render_todo(f, app, &[upper_item_list_area, lower_item_list_area]);
+
+    render_events_list(f, app, event_list_area)
+}
+
+// Renders todo list and info part
+fn render_todo(f: &mut Frame, app: &mut App, area: &[Rect; 2]) {
+    // We create two blocks, one is for the header (outer) and the other is for list (inner).
+    let outer_block = Block::default()
+        .borders(Borders::NONE)
+        .fg(TEXT_COLOR)
+        .bg(TODO_HEADER_BG)
+        .title("TODO List")
+        .title_alignment(Alignment::Center);
+    let inner_block = Block::default()
+        .borders(Borders::NONE)
+        .fg(TEXT_COLOR)
+        .bg(NORMAL_ROW_COLOR);
+
+    // We get the inner area from outer_block. We'll use this area later to render the table.
+    let outer_area = area[0];
+    let inner_area = outer_block.inner(outer_area);
 
-    // Iterate through all elements in the `items` app and append some debug text to it.
+    // We can render the header in outer_area.
+    f.render_widget(outer_block, outer_area);
+
+    // Iterate through all elements in the `items` and stylize them.
     let items: Vec<ListItem> = app
         .items
         .items
         .iter()
-        .map(|i| {
-            let mut lines = vec![Line::from(i.0.bold()).alignment(Alignment::Center)];
-            for _ in 0..i.1 {
-                lines.push(
-                    "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
-                        .italic()
-                        .into(),
-                );
-            }
-            ListItem::new(lines).style(Style::default().fg(Color::Black).bg(Color::White))
+        .enumerate()
+        .map(|(i, (todo, _info, status))| {
```

Comment:
This might be good to extract to a method (which could make sense putting in the item itself)